### DISCUSSION TITLE
Incorporate TotalFreeEnergy with PolycrystalKernelAction

### DISCUSSION THREAD

**Discussion by ashishdhole** [idaholab/moose#27443] [Q&A Modules: Phase field]
Hello!
I am looking for help where I want to compare the total free energy (as calculated by TotalFreeEnergy auxkernel). But I want to take F taken from PolycrystalKernelAction. If you can give the kernel code snippet that return the total energy in this case will also work. I eventually want to put it in my own kernel.
Thank you

**Comment by GiudGiud** [comment]
Hello

The  PolycrystalKernelAction only creates the kernels not the material properties. 
You can look inside the kernels for how they compute (most likely just retrieve) the total free energy
It creates ACGrGrPoly and ACinterface kernels
